cmd/mothership/mothership/start/auditlog.go

```diff
@@ -107,8 +131,50 @@ func auditLogRequest(w http.ResponseWriter, r *http.Request, l *zap.Logger) {
 			return
 		}
 		r.Body = ioutil.NopCloser(bytes.NewBuffer(reqBody))
-		l = l.With(zap.String("requestBody", string(reqBody)))
+		logData.RequestBody = string(reqBody)
+	}
+
+	if ip, ok := os.LookupEnv("POD_IP"); ok {
+		logData.IP = ip
+	} else {
+		o.Logger().Debug("can't find reconciler pod IP")
+	}
+
+	data, err := json.Marshal(logData)
+	if err != nil {
+		server.SendHTTPError(w, http.StatusInternalServerError, &keb.HTTPErrorResponse{
+			Error: errors.Wrap(err, "Failed to marshal auditlog JSON payload").Error(),
+		})
+		return
+	}
+	l.With(zap.String("time", time.Now().Format(time.RFC3339))).
+		With(zap.String("uuid", uuid.New().String())).
+		With(zap.String("user", logData.User)).
+		With(zap.String("data", string(data))).
+		With(zap.String("tenant", o.AuditLogTenantID)).
+		With(zap.String("ip", logData.IP)).
+		With(zap.String("category", "audit.security-events")). // comply with required log backend format
+		Info("")
+}
+
+func getJWTPayload(r *http.Request) (string, error) {
+	jwtHeader := r.Header.Get(XJWTHeaderName)
+	if len(jwtHeader) == 0 {
+		return "", nil
+	}
+	decodedSeg, err := base64.RawURLEncoding.DecodeString(jwtHeader)
```

Comment:
I got 500 internal server error while passing JWT token.
You should use jwt parser or split the token by `.` and then decode it.